edge-agent/test/Microsoft.Azure.Devices.Edge.Agent.Kubernetes.Test/edgedeployment/EdgeDeploymentOperatorTest.cs

```diff
@@ -0,0 +1,210 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Agent.Kubernetes.Test.Edgedeployment
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Net;
+    using System.Net.Http;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using k8s;
+    using k8s.Models;
+    using Microsoft.Azure.Devices.Edge.Agent.Core;
+    using Microsoft.Azure.Devices.Edge.Agent.Kubernetes.EdgeDeployment;
+    using Microsoft.Azure.Devices.Edge.Util.Test.Common;
+    using Microsoft.Rest;
+    using Moq;
+    using Newtonsoft.Json.Linq;
+    using Xunit;
+    using Constants = Microsoft.Azure.Devices.Edge.Agent.Kubernetes.Constants;
+
+    [Unit]
+    public class EdgeDeploymentOperatorTest
+    {
+        const string DeviceNamespace = "a-namespace";
+        const string ExceptionMessage = "ExceptionMessage";
+        static readonly ResourceName ResourceName = new ResourceName("hostname", "deviceId");
+
+        [Fact]
+        public async void SuccessfulEventProcessing()
+        {
+            var status = new EdgeDeploymentStatus(EdgeDeploymentStatusType.Success, "200(OK)");
+            var edgeDefinition = new EdgeDeploymentDefinition("v1", "EdgeDeployment", new V1ObjectMeta(name: ResourceName), new List<KubernetesModule>(), null);
+            var response = new HttpOperationResponse<object>()
+            {
+                Body = edgeDefinition,
+                Response = new HttpResponseMessage(HttpStatusCode.OK)
+            };
+
+            var client = Mock.Of<IKubernetes>();
+            bool mockCallbackCalled = false;
+            Mock.Get(client).Setup(c => c.ReplaceNamespacedCustomObjectStatusWithHttpMessagesAsync(It.IsAny<object>(), Constants.EdgeDeployment.Group, Constants.EdgeDeployment.Version, DeviceNamespace, Constants.EdgeDeployment.Plural, It.IsAny<string>(), null, It.IsAny<CancellationToken>()))
+                .Callback((object o, string group, string version, string _namespace, string plural, string name, Dictionary<string, List<string>> headers, CancellationToken token) =>
+                {
+                    Assert.True(o is JObject);
```

Comment:
It looks like this test verifies too many things: 
1. In a callback it checks that deployment deserialized with the correct status
2. Checks that ReplaceNamespacedCustomObjectStatus method is called

Shouldn't we split into 2 tests?
1. One that will check deserialization into EdgeDeployment with status
2. Work with kube client

If you do this way you don't need callback closure at all. check with `mock.VerifyAll()` to ensure ReplaceNamespacedCustomObjectStatus called without additional bool variables

Reply:
The intent is that the status returned by `DeployModulesAsync()` would be reported in `ReplaceNamespacedCustomObjectStatus()`.

I renamed the test and reorganized it slightly so the intent is clear.